spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/core/DatastoreOperations.java

```diff
@@ -41,11 +41,14 @@
 	/**
 	 * Saves an instance of an object to Cloud Datastore. Behaves as update or insert.
 	 * @param instance the instance to save.
+	 * @return the instance that was saved. This is will have the ID property populated if the
+	 * original saved object had no ID and one was assigned by Cloud Datastore.
 	 */
-	<T> void save(T instance);
+	<T> T save(T instance);
 
 	/**
-	 * Delete an entity from Cloud Datastore.
+	 * Delete an entity from Cloud Datastore. Deleting IDs that do not exist in Cloud Datastore
+	 * is OK.
```

Comment:
"OK" seems ambiguous. Results in a no-op?

Reply:
done.